scripts/ci/cmake/ci-ubuntu20.04-gcc9-mpich.cmake

```diff
@@ -0,0 +1,49 @@
+include(ProcessorCount)
+ProcessorCount(NCPUS)
+math(EXPR N2CPUS "${NCPUS}*2")
+
+set(ENV{CC}  gcc)
+set(ENV{CXX} g++)
+set(ENV{FC}  gfortran)
+
+execute_process(
+  COMMAND "python3-config" "--prefix"
+  OUTPUT_VARIABLE PY_ROOT
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+set(dashboard_cache "
+BUILD_TESTING:BOOL=ON
+ADIOS2_BUILD_EXAMPLES:BOOL=ON
+
+ADIOS2_USE_BZip2:BOOL=ON
+ADIOS2_USE_Blosc:BOOL=ON
+ADIOS2_USE_DataMan:BOOL=ON
+ADIOS2_USE_Fortran:BOOL=ON
+ADIOS2_USE_HDF5:BOOL=ON
+ADIOS2_USE_MPI:BOOL=ON
+ADIOS2_USE_Python:BOOL=ON
+ADIOS2_USE_SZ:BOOL=ON
+ADIOS2_USE_ZeroMQ:STRING=ON
+ADIOS2_USE_ZFP:BOOL=ON
+
+Python_ROOT_DIR:PATH=${PY_ROOT}
+Python_FIND_STRATEGY:STRING=LOCATION
+Python_FIND_FRAMEWORK:STRING=FIRST
+
+CMAKE_C_COMPILER_LAUNCHER=ccache
+CMAKE_CXX_COMPILER_LAUNCHER=ccache
+CMAKE_C_FLAGS:STRING=-Wall
+CMAKE_CXX_FLAGS:STRING=-Wall
+CMAKE_Fortran_FLAGS:STRING=-Wall
+
+MPIEXEC_MAX_NUMPROCS:STRING=${N2CPUS}
+")
+
+# TODO: The Kill* and PreciousTimeStep tests fail (due to timeout) when
+# TODO: adios2 is built "--with-device=ch3:sock:tcp".  Once this is fixed
+# TODO:  in the mpi_dp, we can re-enable these tests.
+set(CTEST_TEST_ARGS EXCLUDE "KillReader|KillWriter|PreciousTimeStep")
```

Comment:
I wonder why didn't this one work though?  See the failed `PreciousTimeStep` tests [here](https://open.cdash.org/viewTest.php?onlyfailed&buildid=9150671).

Reply:
if using | you need to build a regex

Reply:
The Kill* and Precious* tests are long-running tests that involve starting multiple readers over a span of time and killing some of them to try to test failure handling.  Anything that delays startup will impact them more than the other tests that are mostly just writer/reader pairs.

Reply:
> if using | you need to build a regex

Can you spot how what I did above is different than what was already done [here](https://github.com/ornladios/ADIOS2/blob/2d16eb30489e6b8d1aa17a852113e7161155b2cd/scripts/ci/cmake/ci-ubuntu20.04-gcc10-mpich.cmake#L43)?  That looks like a regex.

Reply:
@eisenhauer I'm trying to exclude these tests as a temporary measure so we can start doing most of our builds with mpich.  I found a configuration of mpich that is much faster in CI, with the exception that the `mpi_dp` hangs trying to disconnect when those tests kill a reader or writer.  Until we can come up with a permanent fix for that, we were just planning to exclude those tests from most of the mpich builds (but they should still be tested with the openmpi and icc/oneapi builds).

Reply:
Try something  as "*.(KillReader|KillWriter|PreciousTimeStep).*

Reply:
That doesn't look like a regular expression to me, more like a globbing pattern, but I'll experiment with it and see.  I wish I knew a way to test these locally, instead of pushing and waiting forever on CI.

Reply:
@scottwittenburg you can check them locally by doing ctest -N -R 'regex' (or globbing not sure). I believe that the format is the same